src/d3d11/d3d11_fence.cpp

```diff
@@ -49,14 +58,34 @@ namespace dxvk {
           DWORD               dwAccess,
           LPCWSTR             lpName,
           HANDLE*             pHandle) {
-    Logger::err("D3D11Fence::CreateSharedHandle: Not implemented");
-    return E_NOTIMPL;
+    if (!(m_flags & D3D11_FENCE_FLAG_SHARED))
+      return E_INVALIDARG;
+
+    if (pAttributes)
+      Logger::warn(str::format("CreateSharedHandle: attributes ", pAttributes, " not handled"));
+    if (dwAccess)
+      Logger::warn(str::format("CreateSharedHandle: access ", dwAccess, " not handled"));
+    if (lpName)
+      Logger::warn(str::format("CreateSharedHandle: name ", dxvk::str::fromws(lpName), " not handled"));
+
+    HANDLE sharedHandle = m_fence->sharedHandle();
+    if (sharedHandle == INVALID_HANDLE_VALUE)
+      return E_INVALIDARG;
+
+    *pHandle = sharedHandle;
+    return S_OK;
   }
 
 
   HRESULT STDMETHODCALLTYPE D3D11Fence::SetEventOnCompletion(
           UINT64              Value,
           HANDLE              hEvent) {
+    if (m_flags & D3D11_FENCE_FLAG_SHARED) {
+      // TODO directly call vkWaitSemaphores here, enqueueWait assumes monotonic semaphore
```

Comment:
Why is this an issue? The fence worker should already be able to deal with rewinding semaphores due to the timeout, I don't see any good reason to disable this functionality.

Reply:
I thought it could cause a nasty condition where after the fence is rewound, a wait is started that immediately signals because the timeout hasn't been reached yet.  If this is not too big of a concern for you I'd not be opposed to remove the disable.

Reply:
Rewinding fences is always inherently race-y, we can make the backend a bit more robust there but for now I'd just put a TODO comment there and call it a day. We should definitely enable this functionality.